File: src/irc/band-commands.ts
import { IrcTags } from ".";
import bus from "../message-bus";

export enum Ban {
    ban = 1,
    unband = 2,
}

const mods: string[] = ["yonikosiner", "nniklask"];

export default function bandCommands(message: string, tags: IrcTags): void | boolean {
    if (message.startsWith("!band")) {
        if (!mods.includes(tags["display-name"])) return bus.emit("irc-message", "Only Yoni and mods can banded users");
        bus.emit("band", { message: message, type: Ban.ban });
        return;
    }

    if (message.startsWith("!unband")){
        if (!mods.includes(tags["display-name"])) return bus.emit("irc-message", "Only Yoni and mods can unbanned users");
        bus.emit("band", { message: message, type: Ban.unband });
        return;
    }
};

/*
1A4
1 * 256 = 256
A = 10
10 * 16 = 160
4 * 1  = 4
256 + 160 + 4 = 420
*/

File: pkg/irc/band-commands.go
package irc

import (
	"strings"
)

type Key struct {
    X string
}

func BanCommands(Msg IrcMessage, channel chan IrcMessage) {
    m := map[Key]MessageType{}
    m[Key{"!band"}] = Ban
    m[Key{"!unband"}] = Unban

    banType := m[Key{strings.Fields(Msg.Message)[0]}]
    Msg.Type = MessageType(banType)


    go func() {
        if Msg.Type == Ban || Msg.Type == Unban {
            channel <- Msg
        }
    }()
}
